### PR TITLE
Add cblas128 wrapper

### PR DESCRIPTION
**Pull request by kortschak**
Name to be discussed.

### CODE REVIEW

File: cblas128/cblas128.go [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+// Asum will panic if the vector increment is negative.
+func Asum(n int, x Vector) float64 {
+	if x.Inc < 0 {
+		panic(negInc)
+	}
+	return cblas128.Dzasum(n, x.Data, x.Inc)
+}
+
+// Iamax will panic if the vector increment is negative.
+func Iamax(n int, x Vector) int {
```
Comment (btracey) [inline cblas128/cblas128.go]: This one doesn't panic.

**Comment by btracey** [comment]
The way I interpret it, one has to have the cblas tag in order to use this package, right (at least at the moment)? If it's not there, this file won't exist, and it'll be a compile error.

Except for the one comment, LGTM.

I haven't looked in depth at the complex blas, but I assume there aren't any weird issues that don't appear in blas64.

File: cblas128/cblas128.go (hunk 1 above)
Comment (kortschak) [inline cblas128/cblas128.go]: Also in blas64. Will send PR for that in a sec.

**Comment by kortschak** [comment]
>The way I interpret it, one has to have the cblas tag in order to use this package, right (at least at the moment)? If it's not there, this file won't exist, and it'll be a compile error.

That's correct. Otherwise we have to build OpenBLAS. We have no real tests for this yet, so there's no point building this. I am going through converting the gonum/lapack code to use these and then we have some testing(ish).

**Comment by btracey** [comment]
Unless there's something I'm missing, we should have the same structure in Lapack as we do in BLAS. The thing I'm not sure about with Lapack is what we want to do with the interface. We're never going to have a full go implementation of lapack, and I don't know of any other BLAS distribution that tries. Despite that, we should be able to implement a Go version of linear solve, call it from mat64, and have the go version be the default. I'm not quite sure how to rectify this. Maybe lapack64 only provides convenience wrappers to the functions that have go implementations.

**Comment by kortschak** [comment]
Agreed. I'm not bringing up to that standard in the changes I'm make right now though. I'm just getting it to the state where I can delete gonum/[dz]bw.

**Pull request merged**